### DISCUSSION TITLE
.Net8 and AOT

### DISCUSSION THREAD

**Discussion by KLazarov** [FirelyTeam/firely-net-sdk#2617] [Ideas]
As .Net 8 is being released next week I was wondering if there is any plan to enable AOT for the library as it allows for far faster startup times, which can help a lot for certain use cases.

Cheers!

**Comment by chgl** [comment]
Interested in this as well. Probably the biggest challenge lies with Serialization - at least that's what a very simple AOT test fails with:

```console
System.InvalidOperationException: Reflection-based serialization has been disabled for this application. 
Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.
```

**Reply by ewoutkramer** [reply]
We're using System.Text.Json, and probably some reflection-based serialization, yes (although we've customized it so deeply that I wonder where we are still depending on it). Anyway, our serializers depend deeply on reflection  and IL Emit, which are incompatible with AOT, so the serializers  are out. Which means the FhirClient is out. Those are the most used parts of the SDK, so that's not a great start.

**Comment by ewoutkramer** [comment]
Sorry, I don't know anything about AOT.  What should we do to enable this?  I guess it means we should at least start targeting .NET 8 instead of .NET 6 ;-)
